lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2077,12 +2077,62 @@ bool normalizeWeights(Module *M, ConvolutionNode &CV,
   return true;
 }
 
+/// Gets Constant or returns nullptr if input is not Constant.
+/// Skips QuantizeNode if present.
+static Constant *getConstant(const NodeValue &NV) {
+  Node *N = NV.getNode();
+  if (isa<QuantizeNode>(N)) {
+    N = N->getNthInput(QuantizeNode::InputIdx);
+  }
+  return dyn_cast<Constant>(N);
+}
+
 bool OptimizeBatchNorm::run(Function *F, const CompilationContext &cctx) {
   LOG_SCOPE(F->getLogContext(), getName());
   bool changed = false;
   auto &nodes = F->getNodes();
   auto *M = F->getParent();
 
+  // For each node:
+  for (auto &node : nodes) {
+    // Remove BN if mean,var,eps,scale,beta values make it redundant as per
+    // expression (X - mean) * (1.0 / sqrt(var + eps)) * scale + bias.
+    if (auto *BN = dyn_cast<BatchNormalizationNode>(&node)) {
```

Comment:
nit to reduce indentation
```suggestion
    auto *BN = dyn_cast<BatchNormalizationNode>(&node);
    if (!BN) {
      continue;
    }
```